Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule.py

```diff
@@ -3339,3 +3458,114 @@ def get_dynamic_analysis_command(client: CoreClient, args: Dict) -> CommandResul
         outputs=filtered_alerts,
         raw_response=raw_response,
     )
+
+
+def add_tag_to_endpoints_command(client: CoreClient, args: Dict):
+    endpoint_ids = argToList(args.get('endpoint_ids', []))
+    tag = args.get('tag')
+
+    if any(  # check if any filter arguments provided
+        key in args for key in
+        (
+            'endpoint_id_list', 'dist_name', 'ip_list', 'group_name', 'platform', 'alias_name',
+            'isolate', 'hostname', 'status', 'first_seen_gte', 'first_seen_lte', 'last_seen_gte', 'last_seen_lte'
+        )
+    ):
+        endpoint_id_list = argToList(args.get('endpoint_id_list'))
+        dist_name = argToList(args.get('dist_name'))
+        ip_list = argToList(args.get('ip_list'))
+        group_name = argToList(args.get('group_name'))
+        platform = argToList(args.get('platform'))
+        alias_name = argToList(args.get('alias_name'))
+        isolate = args.get('isolate')
+        hostname = argToList(args.get('hostname'))
+        status = args.get('status')
+
+        first_seen_gte = arg_to_timestamp(
+            arg=args.get('first_seen_gte'),
+            arg_name='first_seen_gte'
+        )
+
+        first_seen_lte = arg_to_timestamp(
+            arg=args.get('first_seen_lte'),
+            arg_name='first_seen_lte'
+        )
+
+        last_seen_gte = arg_to_timestamp(
+            arg=args.get('last_seen_gte'),
+            arg_name='last_seen_gte'
+        )
+
+        last_seen_lte = arg_to_timestamp(
+            arg=args.get('last_seen_lte'),
+            arg_name='last_seen_lte'
+        )
+
+        raw_response = client.add_endpoint_tag(
+            endpoint_ids=endpoint_ids, tag=tag, endpoint_id_list=endpoint_id_list, dist_name=dist_name, ip_list=ip_list,
+            group_name=group_name, platform=platform, alias_name=alias_name, isolate=isolate, hostname=hostname,
+            first_seen_lte=first_seen_lte, first_seen_gte=first_seen_gte,
+            last_seen_lte=last_seen_lte, last_seen_gte=last_seen_gte, status=status
+        )
+    else:
+        raw_response = client.add_endpoint_tag(endpoint_ids=endpoint_ids, tag=tag, no_filter=True)
```

Comment:
why do we need a seperate else case for when they are all not provided? Cant we just pass the None params in and get the empty array?

Reply:
it is not possible to send the request without a filter, so in case no filter was provided, we by default set the `endpoint_ids` provided (as a required argument) as a filter argument for the `endpoint_ids_list`